### ISSUE TITLE
autojump seems to be broken in fish 2.3.0

### ISSUE THREAD

**Issue by hav3n** [wting/autojump#423]
Here's the error I get when cd'ing into any directory in fish 2.3.0:

```
Invalid redirection target: $AUTOJUMP_ERROR_PATH
/usr/local/share/autojump/autojump.fish (line 2):     autojump --add (pwd) >/dev/null 2>>$AUTOJUMP_ERROR_PATH &
                                                                                      ^
in function '__aj_add'
	called on line 5 of file /usr/local/Cellar/fish/2.3.0/share/fish/functions/cd.fish
	with parameter list 'VARIABLE SET PWD'

in event handler: handler for variable 'PWD'
```

And what I get when using j :

```
test: Expected a combining operator like '-a' at index 1
autojump: directory 'Desktop' not found

Please source the correct autojump file in your shell's startup file. For more information, please reinstall autojump and read the post installation instructions.

Try `autojump --help` for more information
```

**Comment by Ramblurr** [comment]
Gah! I thought I was going crazy, then realized fish had gotten an update.

Strangely, if I manually source the script after my shell is booted up it works, see:


```bash
# last line in fish config
# auto jump
[ -f /usr/local/share/autojump/autojump.fish ]; and source /usr/local/share/autojump/autojump.fish
```

```
 somewhere                                                                                                                                                                                    1
test: Expected a combining operator like '-a' at index 1
autojump: directory 'somewhere' not found

Please source the correct autojump file in your shell's startup file. For more information, please reinstall autojump and read the post installation instructions.

Try `autojump --help` for more information.
$ source /usr/local/share/autojump/autojump.fish
$ j somewhere
cd: The directory 'somewhere' does not exist
```

**Comment by wting** [comment]
Should be fixed in v22.3.5 from @hwartig's PR. Please re-open this issue if the bug persists.
